internal/workflow_testsuite.go

```diff
@@ -942,16 +951,28 @@ func (e *TestWorkflowEnvironment) AssertExpectations(t mock.TestingT) bool {
 //
 // It can produce a false result when an argument is a pointer type and the underlying value changed after calling the mocked method.
 func (e *TestWorkflowEnvironment) AssertCalled(t mock.TestingT, methodName string, arguments ...interface{}) bool {
-	return e.mock.AssertCalled(t, methodName, arguments...)
+	dummyT := &testing.T{}
+	if !(e.workflowMock.AssertCalled(dummyT, methodName, arguments...) || e.activityMock.AssertCalled(dummyT, methodName, arguments...)) {
+		return e.workflowMock.AssertCalled(t, methodName, arguments...) && e.activityMock.AssertCalled(t, methodName, arguments...)
+	}
+	return true
 }
 
 // AssertNotCalled asserts that the method was not called with the given arguments.
 // See AssertCalled for more info.
 func (e *TestWorkflowEnvironment) AssertNotCalled(t mock.TestingT, methodName string, arguments ...interface{}) bool {
```

Comment:
I think this fix is acceptable. I don't believe it is backwards compatible technically but if fixing this breaks their test the test was probably wrong to begin with.

Do you think we need workflow and activity versions of this and `AssertNumberOfCalls`  that differentiate between activity and workflow mocks?

Reply:
also  `AssertNumberOfCalls`

Reply:
@cretz in the issue you mention backwards compatibility concerns. Did you have other concerns then these 3 functions?

Reply:
Do you mean `AssertActivityCalled`/`AssertWorkflowCalled` and others? 
We might add them as well with keeping general AssertCalled (and others)

Reply:
> Did you have other concerns then these 3 functions?

That was my only concern I can recall, our high-level assertion calls. I just want to make sure we don't break anyone's existing testing code.

> Do you mean AssertActivityCalled/AssertWorkflowCalled and others?
> We might add them as well with keeping general AssertCalled (and others)

Hrmm, I don't have a strong opinion here. I am guessing we don't want to give `mock.Mock` to the user, so I guess we might as well expose.

Reply:
> I just want to make sure we don't break anyone's existing testing 

I don't know the best approach here, but I can agree with @Quinn-With-Two-Ns that if these changes break anyones test, the test itself was logically wrong

Another point is that temporal allows you to use same names for activity and workflows, but SDK doesn't allow you to test it :)

Reply:
@cretz to be clear I think this change could cause some tests that passed before to fail, but those tests depended on workflow and activity mocks overriding each other which is a bug. I am also not sure how popular these functions even are. I think it is worth the risk of breaking some bad tests to fix this weird footgun for users.

Reply:
Yeah, I definitely don't mind if someone was relying on the name ambiguity and they break. But other than that, if we feel proper tests won't break, I'm good.